### ISSUE TITLE
Save / Load Bugs

### ISSUE THREAD

**Issue by PetrochukM** [pytorch/audio#430]
## 🐛 Bug

The process of saving and loading a file causes it to change significantly up to the 1st decimal point.

## To Reproduce

```python
import torch
import torchaudio

signal = torch.clamp(torch.randn(24000), -1.0, 1.0)
print('Signal Stats:', signal.max(), signal.min(), signal.std(), signal.mean())

torchaudio.save('wavefile.wav', signal, 24000)
loaded_signal, _ = torchaudio.load('wavefile.wav')
print('Loaded Signal Stats:', loaded_signal.max(), loaded_signal.min(), loaded_signal.std(),
      loaded_signal.mean())
```

Example output:
```bash
(venv) OS-XImage-9:Text-to-Speech michaelp$ python foo.py 
Signal Stats: tensor(1.) tensor(-1.) tensor(0.7185) tensor(0.0013)
Loaded Signal Stats: tensor(0.9998) tensor(-1.) tensor(0.6444) tensor(-0.3177)
```

Notice that the mean and standard deviation of the signal has significantly changed.

## Expected behavior

The WAV format should save and load without any loss in precision.

**Comment by PetrochukM** [comment]
Another issue that I found is that if you load a 16-bit PCM file with the load function returns a `torch.Float32` which does not match the `dtype` of the file.

**Comment by vincentqb** [comment]
This appears to be a problem with sox. Changing the backend to `soundfile` fixes the problem
```python
import torch
import torchaudio

torchaudio.set_audio_backend("sox")  # default

signal = torch.clamp(torch.randn(24000), -1.0, 1.0)
print('Signal Stats:', signal.max(), signal.min(), signal.std(), signal.mean())

torchaudio.save('wavefile.wav', signal, 24000)
loaded_signal, _ = torchaudio.load('wavefile.wav')
print('Loaded Signal Stats:', loaded_signal.max(), loaded_signal.min(), loaded_signal.std(), loaded_signal.mean())

# Signal Stats: tensor(1.) tensor(-1.) tensor(0.7190) tensor(0.0057)
# Loaded Signal Stats: tensor(0.9998) tensor(-1.) tensor(0.6426) tensor(-0.3226)

print(torch.norm(signal-loaded_signal, p=1))
# tensor(7880.0034)

torchaudio.set_audio_backend("soundfile")                                                  

signal = torch.clamp(torch.randn(24000), -1.0, 1.0)
print('Signal Stats:', signal.max(), signal.min(), signal.std(), signal.mean())

torchaudio.save('wavefile.wav', signal, 24000)
loaded_signal, _ = torchaudio.load('wavefile.wav')
print('Loaded Signal Stats:', loaded_signal.max(), loaded_signal.min(), loaded_signal.std(), loaded_signal.mean())

# Signal Stats: tensor(1.) tensor(-1.) tensor(0.7162) tensor(0.0028)
# Loaded Signal Stats: tensor(1.0000) tensor(-1.) tensor(0.7162) tensor(0.0028)

print(torch.norm(signal-loaded_signal, p=1))
# tensor(0.3670)
```

**Comment by mthrok** [comment]
I am closing the issue, as in #978, we switched the default backend to `"sox_io"` backend, which does not exhibit this bug.

(The deprecated `"sox"` backend still exhibit this bug but it is scheduled to be removed from the codebase in 0.9.0 release.)
